### PR TITLE
Add storage sink consumer developing guide

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12387
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-storage-consumer-dev-guide.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+## Design a consumer
+
+The following diagram shows the overall consumption process of the consumer:
+
+![TiCDC storage consumer overview](/media/ticdc/ticdc-storage-consumer-overview.png)
+
+The components of the consumer and their features are described as follows:
+
+```
```
Comment (Oreoxmt) [inline ticdc/ticdc-storage-consumer-dev-guide.md:26]:
```suggestion
```go
```

File: ticdc/ticdc-storage-consumer-dev-guide.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+└── test
+    ├── tbl_1
+    │   └── 437752935075545091
+    │       ├── CDC000001.json
+    │       └── schema.json
+```
+
+The consumer parses the table schema of the `schema.json` file and obtains the DDL Query statements:
+
+- If no Query statement is found or `TableVersion` is less than the consumer checkpoint, the consumer skips this step.
```
Comment (Oreoxmt) [inline ticdc/ticdc-storage-consumer-dev-guide.md:113]:
```suggestion
- If no Query statement is found or `TableVersion` is less than the consumer checkpoint, the consumer skips this statement.
```

File: ticdc/ticdc-storage-consumer-dev-guide.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+- If Query statements exist or `TableVersion` is equal to or greater than the consumer checkpoint, the consumer executes the DDL statements in the downstream MySQL.
+
+Then the consumer starts replicating the `CDC000001.json` file.
+
+In the following example, DDL query result in the `test/tbl_1/437752935075545091/schema.json` file is not empty:
```
Comment (Oreoxmt) [inline ticdc/ticdc-storage-consumer-dev-guide.md:118]:
```suggestion
In the following example, the DDL Query statements in the `test/tbl_1/437752935075545091/schema.json` file is not empty:
```

File: ticdc/ticdc-storage-consumer-dev-guide.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            "ColumnName":"OfficeLocation",
+            "ColumnType":"BLOB",
+            "ColumnLength":"20"
+        }
+    ],
+    "TableColumnsTotal":"5"
+}
+```
+
+When the consumer traverses the directory again, it finds another new version directory of the table. Note that you can consume data in the new directory only after all files in the `test/tbl_1/437752935075545091` directory have been consumed.
```
Comment (Oreoxmt) [inline ticdc/ticdc-storage-consumer-dev-guide.md:158]:
```suggestion
When the consumer traverses the directory again, it finds a new version directory of the table. Note that the consumer can consume data in the new directory only after all files in the `test/tbl_1/437752935075545091` directory have been consumed.
```

File: ticdc/ticdc-storage-consumer-dev-guide.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    ├── tbl_1
+    │   ├── 437752935075545091
+    │   │   ├── CDC000001.json
+    │   │   └── schema.json
+    │   └── 437752935075546092
+    │   │   └── CDC000001.json
+    │   │   └── schema.json
+```
+
+The consumption logic is consistent. Specifically, the consumer parses the table schema of the `schema.json` file and obtains and processes DDL query statements accordingly. Then the consumer starts replicating the `CDC000001.json` file.
```
Comment (Oreoxmt) [inline ticdc/ticdc-storage-consumer-dev-guide.md:172]:
```suggestion
The consumption logic is consistent. Specifically, the consumer parses the table schema of the `schema.json` file and obtains and processes DDL Query statements accordingly. Then the consumer starts replicating the `CDC000001.json` file.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**